docs/02.API-REFERENCE.md

```diff
@@ -5030,3 +5234,134 @@ jerry_get_arraybuffer_pointer (const jerry_value_t value);
 **See also**
 
 - [jerry_create_arraybuffer_external](#jerry_create_arraybuffer_external)
+
+
+## jerry_get_typedarray_type
+
+**Summary**
+
+Get the type of the TypedArray.
+
+The returned type is one of the [jerry_typedarray_type_t](#jerry_typedarray_type_t)
+enum value.
+
+**Prototype**
+
+```c
+jerry_typedarray_type_t
+jerry_get_typedarray_type (jerry_value_t value);
+```
+
+- `value` - TypedArray object to query for type.
+- return
+  - the type of the TypedArray
+  - JERRY_TYPEDARRAY_INVALID if the object was not a TypedArray
+
+**Example**
+
+```c
+{
+  jerry_typedarray_type_t expected_type = JERRY_TYPEDARRAY_UINT32;
+  jerry_value_t typedarray = jerry_create_typedarray (expected_klass, 25);
+
+  jerry_typedarray_type_t type = jerry_get_typedarray_type (typedarray);
+
+  // 'type' is now JERRY_TYPEDARRAY_UINT32
+
+  jerry_release_value (typedarray);
+}
+```
+
+**See also**
+
+- [jerry_create_typedarray](#jerry_create_typedarray)
+- [jerry_typedarray_type_t](#jerry_typedarray_type_t)
+
+
+## jerry_get_typedarray_length
+
+**Summary**
+
+Get the element count of the TypedArray as specified during creation.
+
+This is not the same as the byteLength property of a TypedArray object.
+
+**Prototype**
+
+```
+jerry_length_t
+jerry_get_typedarray_length (jerry_value_t value);
+```
+
+- `value` - TypedArray object to query
+- return
+  - length (element count) of the TypedArray object
+  - 0 if the object is not a TypedArray
+
+**Example**
+
+```c
+{
+  jerry_value_t array = jerry_create_typedarray (JERRY_TYPEDARRAY_INT32, 21);
+
+  jerry_length_t element_count = jerry_get_typedarray_length (array);
+
+  // element_count is now 21.
+
+  jerry_release_value (array);
+}
+```
+
+**See also**
+
+- [jerry_create_typedarray](#jerry_create_typedarray)
+
+
+## jerry_get_typedarray_buffer
+
+**Summary**
+
+Get the ArrayBuffer object used by a TypedArray object.
+Additionally returns the byteLength and byteOffset properties
+of the TypedArray object.
+
+For the returned ArrayBuffer the [jerry_release_value](#jerry_release_value)
+must be called.
+
+**Prototype**
+
+```c
+jerry_value_t jerry_get_typedarray_buffer (jerry_value_t value,
+                                           jerry_length_t *byteOffset,
+                                           jerry_length_t *byteLength);
+```
+
+- `value` - TypedArray to get the ArrayBuffer from
+- `byteOffset` - returns the start offset of the ArrayBuffer for the TypedArray
+- `byteLength` - returns the number of bytes used from the ArrayBuffer for the TypedArray
```

Comment:
Please mention that these are optional arguments.

Reply:
I've update the PR with the "optional" markings for the parameters.